lib/_http_server.js

```diff
@@ -482,8 +483,15 @@ function connectionListener(socket) {
 
       res.detachSocket(socket);
 
-      if (res._last) {
-        socket.destroySoon();
+      if (res._last) socket.destroySoon();
+      else if (outgoing.length === 0) {
+        // When outgoing queue is empty we will wait for
+        // server.keepAliveTimeout and destroy socket not emiting
+        // 'timeout' events in req, res and server objects
+        socket.removeAllListeners('timeout');
+        socket.setTimeout(self.keepAliveTimeout, function onKeepAliveTimeout() {
```

Comment:
I think we have different understanding of connection closed successfully. As I understand it - properly closed connection scenario should look like this:

1. Open TCP connection
2. Send request
3. Get response
4. Close TCP connection

This PR has nothing to do with this scenario, though, as it tries to handle:

1. Open TCP connection
2. Send request
3. Get response
4. Let's wait for some time without closing the socket, perhaps we will use it later...

The case we are handling here is definitely a timeout case, and if the users want to kill the connection - sending 408 on it totally makes sense.